hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusQueueProcessors.java

```diff
@@ -0,0 +1,180 @@
+import org.hawkular.cmdgw.command.ws.server.WebSocketHelper;
+
+@Startup
+@Singleton
+@TransactionAttribute(TransactionAttributeType.NOT_SUPPORTED)
+public class BusQueueProcessors {
+
+    private static class FeedQueueListener<T extends BasicMessage>
+            extends org.hawkular.bus.common.consumer.AbstractBasicMessageListener<T> {
+        private final ConnectedSessions connectedFeeds;
+
+        public FeedQueueListener(ConnectedSessions connectedFeeds) {
+            super();
+            this.connectedFeeds = connectedFeeds;
+        }
+
+        @Override
+        public void onMessage(Message message) {
+            try {
+                String feedId = message.getStringProperty(Constants.HEADER_FEEDID);
+                String basicMessageClass = message.getStringProperty(Constants.HEADER_BASIC_MESSAGE_CLASS);
+                Class<T> cl = (Class<T>) Class.forName(basicMessageClass);
+                if (feedId != null) {
+                    Session session = connectedFeeds.getSession(feedId);
+                    BasicMessageWithExtraData<T> request = parseMessage(message, cl);
+                    new WebSocketHelper().sendSync(session, request);
+                } else {
+                    // TODO send a message back to the UI saying that there is no such feed
+                }
+            } catch (Exception e) {
+                MsgLogger.LOG.errorCannotProcessExecuteOperationResponseMessage(e);
+            }
+        }
+
+    }
+
+    private static class UiQueueListener<T extends BasicMessage>
+            extends org.hawkular.bus.common.consumer.AbstractBasicMessageListener<T> {
+        private final CommandContextFactory commandContextFactory;
+        private final Commands commands;
+
+        public UiQueueListener(CommandContextFactory commandContextFactory, Commands commands) {
+            super();
+            this.commandContextFactory = commandContextFactory;
+            this.commands = commands;
+        }
+
+        @Override
+        public void onMessage(Message message) {
+            try {
+                String basicMessageClass = message.getStringProperty(Constants.HEADER_BASIC_MESSAGE_CLASS);
+                Class<T> cl = (Class<T>) Class.forName(basicMessageClass);
+                BasicMessageWithExtraData<T> messageWithData = parseMessage(message, cl);
+                T request = messageWithData.getBasicMessage();
+                Command<BasicMessage> command = (Command<BasicMessage>) commands.getCommand(request.getClass());
+                CommandContext context = commandContextFactory.newCommandContext(null);
+                command.execute(request, messageWithData.getBinaryData(), context);
+            } catch (Exception e) {
+                MsgLogger.LOG.errorCannotProcessExecuteOperationResponseMessage(e);
+            }
+        }
+
+        protected void onBasicMessage(BasicMessageWithExtraData<T> messageWithData) {
+        }
```

Comment:
That's actually how the original code was. The listener only accepted classes as defined in the signature of the listener (via generics). But, also, keep in mind this is just the command POJO - the JSON is parsed into this Command class. The execution code is done elsewhere, and only those command classes that that are prebuilt inside command-gateway would work. Its not like you can send in a phony JSON and somehow get it to deserialize into a malicious command. At least, I can't see how that could be - but then, I'm not an evil person so I have a hard time thinking how this can be defeated. :)